internal/sink/buffer.go

```diff
@@ -0,0 +1,41 @@
+package sink
+
+import (
+	"sync"
+)
+
+type SinkBuffer struct {
+	mu   sync.Mutex
+	data map[string]SinkMessage
+}
+
+func NewSinkBuffer() SinkBuffer {
+	return SinkBuffer{
+		data: map[string]SinkMessage{},
+	}
+}
+
+func (b *SinkBuffer) Size() int {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	return len(b.data)
+}
+
+func (b *SinkBuffer) Add(message SinkMessage) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	// Unique identifier for each message (topic + partition + offset)
+	key := message.KafkaMessage.String()
```

Comment:
Can we introduce a method for this on `SinkMessage`? Kafka appearing here doesn't read well.

Reply:
This is actually the Kafka Message's key, not the event/payload.